libs/trucking-kit/company/data-access/src/lib/company.model.ts

```diff
@@ -0,0 +1,19 @@
+import { Company } from '@shared/models';
+
+export type CompanyState = {
+  companies: Company[];
+  creating: boolean;
+  fetching: boolean;
+  loading: boolean;
+  updating: boolean;
+  removing?: string;
+  selected?: Company;
```

Comment:
Perhaps `selected` should just be the identifier of a selected company, so the dot number? Then you can derive a find, instead of having a duplicate company object on that state?